interface/global-preferences (#30)
cgi-studio-gmbh/incari-doc: interface/global-preferences.md

# Global Preferences 

> **Global Preferences** allows us to apply default settings for aspects of our Incari **Project**. These aspects are discussed below:

Reviewer: "Incari **Project**" - This is misleading because the preferences are global and affect every project, not just the current one.

{% tabs %}

{% tab title="Animation Editor" %}
![](../.gitbook/assets/interface/global-preferences/animation-editor.png)
{% endtab %}

## Playback
### Loop Mode 
`Loop Mode` determines if an animation should play forward or alternate between playing forward and backward in a **Scene**. It can be set in the following ways:
* `Alternate`: The animation is played forward and backward in each animation cycle.

> * `None`: The animation is not played.

Reviewer: Not looped! As in it plays once.

There are some animated GIFs to illustrate the looping behaviors in the assets folder:

![image](https://user-images.githubusercontent.com/20472837/91956060-65b16c80-ed04-11ea-9c4c-3c4472f67e93.png)

* `Repeat`: The animation is played forward in repeated cycles without reversing direction.

{% tab title="Global" %}
![](../.gitbook/assets/interface/global-preferences/global.png)
{% endtab %}

## Projects
### Projects Path
The default directory on a user's computer where Incari **Projects** are saved. 

{% tab title="Logic Editor" %}
![](../.gitbook/assets/interface/global-preferences/logic-editor.png)
{% endtab %}

## Theme
### Grid Color 
The `Grid color` of the **Logic Editor** can be set here. The *color* can be set by manually adjusting the the **HSB** / **RGBA** values or inserting a **Hex Code**, using their corresponding [**Color Attribute**](../_archive/attributes/attribute-types/color-attributes.md).

Also, the `Reset color` button sets the `Grid color` back to its default value.

{% tab title="User Interface" %}
![](../.gitbook/assets/interface/global-preferences/user-interface.png)
{% endtab %}

## Layout 
### Reset layout

> The `Reset Layout` button resets **Interface**s found in Incari to its default positions.

Reviewer: ```suggestion
The `Reset Layout` button resets the positioning and visibility of all **Modules** to their defaults.
```

### Grid Color
`Grid Color` sets the color of the *grid* found in the **ViewPort**. The color is set in the same way as `Bottom` and `Top Gradient Color` **Attributes**.

`Reset colors` sets the `Bottom` and `Top Gradient Color` as well as the `Grid Color` of the **Viewport** back to their default colors. 

> In order to open the **Global Preference** **Interface**, please the see the illustration below:

Reviewer: As far as I know, we are using the term **Module** to describe all of the following:
![image](https://user-images.githubusercontent.com/20472837/91955436-757c8100-ed03-11ea-8800-39c5bc93c885.png)
The word "interface" is used more for the overall GUI of Incari itself.
